lib/src/components/component.dart

```diff
@@ -46,6 +46,10 @@ abstract class Component {
   /// Renders this component on the provided Canvas [c].
   void render(Canvas c) {}
 
+  /// This is used to render this component and potential children on subclasses
+  /// of [Component] on the provided Canvas [c].
```

Comment:
you probably should not override this method, as it's already done by BaseComponent

Reply:
You mean that I should add that as a comment?

Reply:
I think it should be fine, since it has `@mustCallSuper` on `BaseGame` and if they are down doing implementation on a `Component` level they hopefully know what they are doing.